In [12]:
import numpy as np
from math import factorial

**Задача №1.** Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800]. Найдите ее среднее значение и дисперсию.

Для определения среднего значения и дисперсии непрерывной СВ А на промежутке (200, 800] будем использовать формулы нахождения математического ожидания:<br> $$ M(X)=\frac{a+b}{2} $$ и дисперсии: $$ D(X)=\frac{(b-a)^2}{12} $$

In [35]:
def avg_A (a, b):
    avg_A = (a + b) / 2
    return avg_A

In [36]:
avg_A (200, 800)

500.0

**Ответ:** среднее значение непрерывной СВ А = **500**

In [9]:
def variance(a, b):
    d = (b - a)**2 / 12
    return d

In [10]:
variance(200, 800)

30000.0

**Ответ:** дисперсия непрерывной СВ А = **30000**

**Задача №2.** О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

Для определения правой границы непрерывной СВ В воспользуемся формулой нахождения дисперсии, и выведим из данной формулы формулу нахождения правой границы b.
$$ D(X)=\frac{(b-a)^2}{12} $$
где:<br>  
дисперсия D(X) = 0.2, <br>
левая граница a = 0.5 <br>  
Найдём b: <br>  
$ 0.2=\frac{(b-0.5)^2}{12} $  
$ b=0.5+\sqrt{2.4} $  



In [21]:
b = round((a + 2.4 ** (1/2)), 4)

In [22]:
b

2.0492

**Ответ:** значение правой границы непрерывной СВ В **b= 2.0492**


Среднее значение непрерывной СВ В найдем по формуле нахождения математического ожидания:<br> $$ M(X)=\frac{a+b}{2} $$

In [33]:
def avg_B (a, b):
    avg_B = (a + b) / 2
    return avg_B

In [34]:
avg_B (0.5, 2.0492)

1.2746

**Ответ:** среднее значение непрерывной СВ В **avg_B = 1.2746**

**Задача №3.** Непрерывная случайная величина X распределена нормально и задана плотностью распределения: <br> $$f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)$$
Найдите:<br>
а) M(X)<br>
б) D(X)<br>
в) std(X) (среднее квадратичное отклонение)

Для нахождения математического ожидания M(x), дисперсии D(x), среднего квадратичного отклонения std(X) воспользуемся общей формулой плотности нормального распределения:  
$ f(x)=\frac{1}{4*\sqrt{2\pi}}*e^{-\frac{(x+2)^2}{32}} $  
M(X) = a = -2  
$ \sigma=4 $  
$ D(X)=\sigma^2=16 $

**Ответ:**  
M(X) = - 2  
D(X) = 16  
$ \sigma=4 $

**Задача №4.** Рост взрослого населения города X имеет нормальное распределение.
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.

Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост: <br>
а) больше 182 см <br>
б) больше 190 см <br>
в) от 166 см до 190 см <br>
г) от 166 см до 182 см <br>
д) от 158 см до 190 см <br>
е) не выше 150 см или не ниже 190 см <br>
ё) не выше 150 см или не ниже 198 см <br>
ж) ниже 166 см.

Воспользуемся формулой расчёта Z-значения:<br>
$$ Z=\frac{X-μ}{\sigma}$$

In [47]:
mu = 174
sigma = 8

In [48]:
def z(x):
    z = (x - mu) / sigma
    return z

Для расчёта вероятности будем использовать Z-таблицу

**а) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: больше 182 см**

In [58]:
z_a = z(182)

In [59]:
z_a

1.0

По Z-таблице для z_a =1.0 вероятность **р = 0.84134**

In [62]:
 p = 0.84134

In [65]:
p_a = round((1 - p), 5)

In [66]:
p_a

0.15866

**Ответ: р_a = 0.15866**

**б) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: больше 190 см**

In [51]:
z_b = z(190)

In [52]:
z_b

2.0

По Z-таблице для z_b =2.0 вероятность **р = 0.97725**

In [67]:
p = 0.97725

In [68]:
p_b = round((1 - p), 5)

In [69]:
p_b

0.02275

**Ответ: р_b = 0.02275**

**в) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: от 166 см до 190 см**


In [73]:
z_1 = z(166)

In [74]:
z_1

-1.0

По Z-таблице для z_1 = -1.0 вероятность **р1 = 0.15866**

In [83]:
p1= 0.15866

In [77]:
z_2 = z(190)

In [79]:
z_2

2.0

По Z-таблице для z_2 =2.0 вероятность **р2 = 0.97725**

In [86]:
p2 = 0.97725

In [89]:
p_c = round((p2-p1), 5)

In [90]:
p_c

0.81859

**Ответ: р_c = 0.081859**

**г) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: от 166 см до 182 см**

In [100]:
z_1 = z(166)

In [101]:
z_1

-1.0

По Z-таблице для z_1 = -1.0 вероятность **р1 = 0.15866**

In [102]:
p1 = 0.15866

Значение p2 примем из выше приведенного решения в пункте а) **р2=0.84134**

In [103]:
p2 = 0.84134

In [104]:
p_d = round((p2-p1), 5)

In [105]:
p_d

0.68268

**Ответ: р_d = 0.68268**

**д) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: от 158 см до 190 см**

In [106]:
z_1 = z(158)

In [107]:
z_1

-2.0

По Z-таблице для z_1 = -2.0 вероятность **р1 = 0.02275**

In [110]:
p1 =0.02275

Значение p2 примем из выше приведенного решения в пункте в) **р2 = 0.97725**

In [111]:
p2 = 0.97725

In [112]:
p_e = round((p2-p1), 5)

In [113]:
p_e

0.9545

**Ответ: р_e = 0.9545**

**е) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: не выше 150 см или не ниже 190 см**

In [114]:
z_1 = z(150)

In [119]:
z_1

-3.0

По Z-таблице для z_1 = -3.0 вероятность **р1 = 0.00135**

In [120]:
p1 = 0.00135

Значение p2 примем из выше приведенного решения в пункте в) **р2 = 0.97725**

In [121]:
p2 = 0.97725

In [127]:
p_f = round((p1+(1-p2)), 5)

In [128]:
p_f

0.0241

**Ответ: р_f = 0.0241**

**ё) Вероятность того, что случайным образом выбранный взрослый человек имеет рост: не выше 150 см или не ниже 198 см**

In [129]:
z_1 = z(150)

In [130]:
z_1

-3.0

По Z-таблице для z_1 = -3.0 вероятность **р1 = 0.00135**

In [131]:
p1 = 0.00135

In [132]:
z_2 = z(198)

In [133]:
z_2

3.0

По Z-таблице для z_2 = 3.0 вероятность **р2 = 0.99865**

In [134]:
p2 = 0.99865

In [135]:
p_g = round((p1+(1-p2)), 5)

In [137]:
p_g

0.0027

**Ответ: р_g = 0.0027**

**ж) Вероятность того, что случайным образом выбранный взрослый человек имеет рост:  ниже 166 см**

In [141]:
z = z(166)

In [142]:
z

-1.0

По Z-таблице для z = -1.0 вероятность **p_h = 0.15866**

In [140]:
p_h = 0.15866

**Ответ: р_h = 0.15866**

**Задача №5.**  На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Для определения на сколько рост человека отклоняется от математического ожидания воспользуемся формулой расчёта Z-значения:<br>
$ Z=\frac{X-μ}{\sigma}$

In [37]:
z = (190 - 178)/(25 ** (1/2))

In [38]:
z

2.4

**Ответ:** Рост человека отклоняется от математического ожидания на **2.4 сигм**